install-ci-tools.sh

```diff
@@ -200,8 +200,46 @@ EOF
     ranch)
       if [ ! -x ranch ] || [ "$(./ranch version)" != "$version" ]; then
         rm -f ranch
-        curl -fsSL http://ranch-updates.goodeggs.com/stable/ranch/${version}/linux-amd64.gz | gunzip > ranch
+        curl -fsSL "http://ranch-updates.goodeggs.com/stable/ranch/${version}/linux-amd64.gz" | gunzip > ranch_real
+        chmod +x ranch_real
+      cat > ranch <<"EOF"
+#!/bin/bash
+set -e
+set -o pipefail
+
+# Make sure and clean up
+trap "exit" INT TERM ERR
+trap "kill 0" EXIT
```

Comment:
does `kill 0` clean up child processes?  can you explain these two lines a little more? ;)

Reply:
It does indeed!  `man 2 kill` "If pid equals 0, then sig is sent to every process in the process group of the calling process."